app/models/ingestible.rb

```diff
@@ -158,4 +164,19 @@ def update_buffers
     self.works_buffer_updated_at = Time.current
     save
   end
+
+  def locked?
+    locked_at.present? && locked_at > LOCK_TIMEOUT_IN_SECONDS.seconds.ago
+  end
+
+  def obtain_lock(user)
+    return false if locked? && locked_by_user_id != user.id
+
+    update!(locked_at: Time.zone.now, locked_by_user: user)
+    return true
+  end
+
+  def release_lock
```

Comment:
I've added this method, but actually I don't found proper place to use it.
So for now all locks will stay active for 15 minutes after last access

Reply:
Yeah, maybe we would implement an admin-override button that would force a lock open. Good planning ahead.